File: logdemo/src/main/java/ua/at/tsvetkov/demo/AppTaoLogDemo.java
package ua.at.tsvetkov.demo;

import android.app.Application;

import ua.at.tsvetkov.util.Log;

/**
 * Created by lordtao on 06.12.2017.
 */

public class AppTaoLogDemo extends Application {

   @Override
   public void onCreate() {
      super.onCreate();

      Log.setStamp(BuildConfig.GIT_SHA);
      Log.enableComponentsChangesLogging(this);

   }

}

File: demokotlin/src/main/java/ua/at/tsvetkov/demokotlin/AppTaoLogDemo.kt
package ua.at.tsvetkov.demokotlin

import android.app.Application
import ua.at.tsvetkov.util.logger.Log.setDisabled
import ua.at.tsvetkov.util.logger.LogComponents.enableComponentsChangesLogging

/**
 * Created by lordtao on 06.12.2017.
 */
class AppTaoLogDemo : Application() {

    override fun onCreate() {
        super.onCreate()
        if (BuildConfig.DEBUG) {
            enableComponentsChangesLogging(this)
        } else {
            setDisabled()
        }
    }

}